### PR TITLE
IMU support

### PR DESCRIPTION
**Pull request by rbolboac**
- added generic support for IMUs
- added driver for adis16505
- added examples for adis16505

### CODE REVIEW

File: drivers/imu/iio_adis.c [hunk 1]
```diff
@@ -0,0 +1,1414 @@
...
+	if (!iio_adis->adis_dev)
+		return -EINVAL;
+
+	adis = iio_adis->adis_dev;
+
+	switch (channel->ch_num) {
+	case ADIS_GYRO_X:
+		ret = adis_read_xg_bias(adis, &res);
+		if (ret)
+			return ret;
```
Comment (CiprianRegus) [inline drivers/imu/iio_adis.c:158]: You can move this if outside the `switch`.

File: drivers/imu/iio_adis.c [hunk 2]
```diff
@@ -0,0 +1,1414 @@
...
+int iio_adis_read_rang_mdl(struct adis_dev* adis, char *buf)
+{
+	int ret;
+	unsigned int rang_mdl;
+	size_t len;
+	ret = adis_read_rang_mdl(adis, &rang_mdl);
+	if (ret)
+		return ret;
+
+	len = snprintf(buf, MAX_TXT_SIZE, rang_mdl_txt[rang_mdl]);
```
Comment (CiprianRegus) [inline drivers/imu/iio_adis.c:570]: return directly

File: drivers/imu/adis.c [hunk 3]
```diff
@@ -0,0 +1,2070 @@
...
+		msgs_to_send[send_idx] = msgs[2];
+		msgs_to_send[send_idx + 1] = msgs[3];
+		send_idx += 2;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	ret = no_os_spi_transfer(adis->spi_desc, msgs_to_send, send_idx);
+	if (ret < 0)
```
[the remark this answers is not in the document]
Comment (rbolboac) [reply drivers/imu/adis.c:314]: this is actually related to linux platform.... linux_spi_transfer will actually return a positive value upon success, that is why if (ret) would not work in this case

File: drivers/imu/adis.c [hunk 4]
```diff
@@ -0,0 +1,2070 @@
...
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	for (i = 1; i <= size; i++, send_idx ++)
+		msgs_to_send[send_idx] = msgs[i];
+
+	ret = no_os_spi_transfer(adis->spi_desc, msgs_to_send, send_idx);
+	if (ret < 0)
```
[the remark this answers is not in the document]
Comment (rbolboac) [reply drivers/imu/adis.c:412]: same as above

File: drivers/imu/adis.c [hunk 5]
```diff
@@ -0,0 +1,2075 @@
...
+
+	if(data->has_paging)
+		dev->current_page = -1;
+	else
+		dev->current_page = 0;
+
+	ret = no_os_gpio_get_optional(&dev->gpio_reset, data->adis_ip->gpio_reset);
+	if (ret) {
+		pr_warning("No reset pin found \n");
+		goto error_gpio;
```
Comment (CiprianRegus) [inline drivers/imu/adis.c:120]: Should this be an error?

Comment (rbolboac) [reply drivers/imu/adis.c:120]: no, it should be just a warning and should continue with initialization, will modify, thanks!

**Review by buha — APPROVED** [review]

**Comment by rbolboac** [comment]
- fixed maxim build
- changed scan mode to big endian false

**Pull request merged**